### ISSUE TITLE
[Train] Ray Train should support AWS trainium instances

### ISSUE THREAD

**Issue by gilvikra** [ray-project/ray#33504]
### Description

I would like AWS trainium instances requiring "xla" torch backend be supported with ray.

  [https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuronx/tutorials/training/distributed_data_parallel.html#neur[…]rial](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuronx/tutorials/training/distributed_data_parallel.html#neuronx-ddp-tutorial)

There is a great push towards Trainium and right now ray does not  seem to support it natively like CPU and GPUs

### Use case

Use of AWS Trainium chips for efficient, performant, cost effective distributed training  on top of ray.

**Comment by swaroopch** [comment]
+1. Given the shortage of GPUs in the industry, it would be beneficial for us to have Ray tested and supported on AWS Trainium, to unblock LLM use cases.

**Comment by pdames** [comment]
Follow-up issue: https://github.com/ray-project/ray/issues/38473. This improves the maintainability of https://github.com/ray-project/enhancements/pull/39 by removing the need to continuously update a hard-coded dictionary of EC2 instance types to neuron core counts.

**Comment by anyscalesam** [comment]
@woshiyyya can you take a look; I'm adding triage as well in case we want to punt this to the next on-call rotation.

**Comment by woshiyyya** [comment]
OK. Will take a look at the CI issue of #39130.

**Comment by matthewdeng** [comment]
Closing this issue as support has since been added - example [here](https://docs.ray.io/en/releases-2.48.0/train/examples/aws-trainium/llama3.html).
